### PR TITLE
Resolves #9820 use-database-names for Scaffolding DBContext columns

### LINKED ISSUE #9820: UseDatabaseNames not working in columns names
I use the flag --use-database-names when generating my classes, table names are done fine, but field names are still PascalCaseify.

Regards!

### PR DESCRIPTION
**Pull request by chrisAmelinckx**
**Please check if the PR fulfills these requirements**

- [x] The code builds and tests pass (verified by our automated build checks)
- [x] Commit messages follow this format
```
    Summary of the changes
    - Detail 1
    - Detail 2

    Fixes #bugnumber
```
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Code meets the expectations our engineering guidelines. https://github.com/aspnet/Home/wiki/Engineering-guidelines.

Please review the guidelines for CONTRIBUTING.md for more details.

### CODE REVIEW

**Comment by smitpatel** [comment]
@chrisAmelinckx - Can you add a test in `RelationalDatabaseModelFactoryTest`

**Comment by chrisAmelinckx** [comment]
@smitpatel Added the tests

File: test/EFCore.Design.Tests/Scaffolding/Internal/RelationalScaffoldingModelFactoryTest.cs [hunk 1]
```diff
@@ -198,6 +198,118 @@ public void Loads_column_types()
...
+                                Name = "Vendor_Discount",
+                                StoreType = "nvarchar(max)"
+                            }
+                        },
+                        PrimaryKey = IdPrimaryKey
+                    }
+                }
+            };
+
+            var entityType = (EntityType)_factory.Create(info, useDatabaseNames: true).FindEntityType("NaturalProducts");
```
Comment (smitpatel) [inline test/EFCore.Design.Tests/Scaffolding/Internal/RelationalScaffoldingModelFactoryTest.cs]: You wouldn't need cast to EntityType. `IEntityType` should be fine to use.

Comment (chrisAmelinckx) [reply test/EFCore.Design.Tests/Scaffolding/Internal/RelationalScaffoldingModelFactoryTest.cs]: Good point, changed it.

**Comment by smitpatel** [comment]
@chrisAmelinckx - Can you squash all commits into 1 commit?

**Comment by smitpatel** [comment]
Thank you for contribution @chrisAmelinckx

**Comment by ErikEJ** [comment]
@smitpatel What is milestone for this?

**Comment by ajcvickers** [comment]
@ErikEJ This is for #9820 which is in 2.1

**Comment by smitpatel** [comment]
@chakrabar - The whole pipeline of processing database name to identifier is at present bit messy. There are multiple components involved - use-database-names/changing case, pluralizer, valid & unique identifier. We will simplify the code while cleaning up that.

**Comment by chakrabar** [comment]
Cool, that works. Any tentative plan for that cleaning up work?

**Comment by smitpatel** [comment]
@chakrabar - #8853 is the issue tracking it and planned for 2.2

**Comment by max-favilli** [comment]
Seems to work with 2.1 RC1 now

Only thing, you need to install packages on the startup projects too, installing them on the target project is not enough.

**Pull request merged**